mirrord/config/src/container.rs

```diff
@@ -1,11 +1,44 @@
-use std::{net::IpAddr, path::PathBuf};
+use std::{fmt, net::IpAddr, path::PathBuf};
 
+use clap::ValueEnum;
 use mirrord_config_derive::MirrordConfig;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 use crate::config::source::MirrordConfigSource;
 
+/// Container runtimes supported by mirrord.
+#[derive(
+    Clone, Copy, Debug, Eq, PartialEq, JsonSchema, Serialize, Deserialize, Default, ValueEnum,
+)]
+#[serde(rename_all = "lowercase")]
+pub enum ContainerRuntime {
+    /// Docker container runtime. (default)
+    #[default]
```

Comment:
I just would check it doesn't affect the container command, just that it still works and does a correct error message if you do some unsupported runtime (like that it doesn't auto default because of clap logic or something)